server/src/main/java/org/opensearch/search/internal/ContextIndexSearcher.java

```diff
@@ -332,6 +317,9 @@ private void searchLeaf(LeafReaderContext ctx, Weight weight, Collector collecto
                 } catch (CollectionTerminatedException e) {
                     // collection was terminated prematurely
                     // continue with the following leaf
+                } catch (QueryPhase.TimeExceededException e) {
+                    searchContext.setSearchTimedOut(true);
```

Comment:
I believe `QueryPhase::searchWithCollector` would need adjustment, it is still driven by `TimeExceededException`, right?

Reply:
Yep that's handled in this PR here: https://github.com/opensearch-project/OpenSearch/pull/8306/files#diff-c619a41ca6b98204ba89f2705e6db9fb8dc145c63dd42b9197a198495def4243

Reply:
Apologies, I missed that somehow